src/client/components/MainForm.tsx

```diff
@@ -42,23 +58,111 @@ export const MainForm = ({
   submitButtonText,
   submitButtonPriority = 'primary',
   submitButtonHalfWidth,
-  useRecaptcha = false,
+  recaptchaSiteKey,
+  setRecaptchaErrorMessage,
+  setRecaptchaErrorContext,
   hasGuardianTerms = false,
   onSubmitOverride,
-}: PropsWithChildren<Props>) => {
-  const hasTerms = useRecaptcha || hasGuardianTerms;
+}: PropsWithChildren<MainFormProps>) => {
+  const recaptchaEnabled = !!recaptchaSiteKey;
+  const hasTerms = recaptchaEnabled || hasGuardianTerms;
+
+  const registerFormRef = createRef<HTMLFormElement>();
+  const [recaptchaState, setRecaptchaState] =
+    useState<UseRecaptchaReturnValue>();
+
+  /**
+   * Executes the reCAPTCHA check.
+   * Prevents the form from submitting until the reCAPTCHA check is complete.
+   */
+  const handleSubmit = useCallback(
+    (event: React.FormEvent<HTMLFormElement>) => {
+      if (onSubmitOverride) {
+        onSubmitOverride(event);
+      }
+
+      if (recaptchaEnabled && !recaptchaState?.token) {
+        event.preventDefault();
+        recaptchaState?.executeCaptcha();
+      }
+    },
+    [onSubmitOverride, recaptchaEnabled, recaptchaState],
+  );
+
+  /**
+   * Submits the form once the reCAPTCHA check has been successfully completed.
+   */
+  useEffect(() => {
+    if (recaptchaEnabled) {
+      const registerFormElement = registerFormRef.current;
+      if (recaptchaState?.token) {
+        registerFormElement?.submit();
+      }
+    }
+  }, [
+    recaptchaEnabled,
+    recaptchaState,
+    recaptchaState?.token,
+    registerFormRef,
```

Comment:
I don't think changes to `registerFormRef` will trigger a rerender
https://stackoverflow.com/a/60476525

Reply:
Interesting, good spot! I'll read into it.

Reply:
after looking into this, I don't think this is too critical to fix, as we don't change the ref after rendering the page. it's a pretty static ref just used to trigger submit on the form

Reply:
> after looking into this, I don't think this is too critical to fix, as we don't change the ref after rendering the page. it's a pretty static ref just used to trigger submit on the form

It would be good to remove it from the list of dependencies unless there's a reason for it being there. It's a bit strange seeing a ref as a useEffect dependency 🤔

Reply:
I don't think it should matter, yeah - I've removed it and added a comment above explicitly ignoring it so the hook linter passes https://github.com/guardian/gateway/pull/1176/commits/b44b46dfe6bb3162ef4003e40701a9ff5f1398e6#

Reply:
ah if `exhaustive-deps` complains about this, then we're probably supposed to do it this way
sorry for making such a fuss

Reply:
no problem, it was a good thing to point out regardless